src/app/component/user/modal-auth/windowsignin/windowsignin.component.ts

```diff
@@ -0,0 +1,134 @@
+import { Component, Inject, OnInit, ViewEncapsulation } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+import { UserOwnSignInService } from '../../../../service/auth/user-own-sign-in.service';
+import { UserOwnSignIn } from '../../../../model/user-own-sign-in';
+import { HttpErrorResponse } from '@angular/common/http';
+import { UserSuccessSignIn } from '../../../../model/user-success-sign-in';
+import { Router } from '@angular/router';
+import { AuthService, GoogleLoginProvider } from 'angularx-social-login';
+import { GoogleSignInService } from '../../../../service/auth/google-sign-in.service';
+import { MatDialog } from '@angular/material';
+import { RestoreComponent } from '../../restore/restore.component';
+import { LocalStorageService } from '../../../../service/localstorage/local-storage.service';
+//import { WindowsignupComponent } from 'src/app/component/user/modal-auth/windowsignup/windowsignup.component';
+
+@Component({
+  selector: 'app-windowsignin',
+  templateUrl: './windowsignin.component.html',
+  styleUrls: ['./windowsignin.component.css']
+})
+export class WindowsigninComponent implements OnInit {
+  userOwnSignIn: UserOwnSignIn;
+  loadingAnim: boolean;
+
+  emailErrorMessageBackEnd: string;
+  passwordErrorMessageBackEnd: string;
+
+  backEndError: string;
+
+  constructor(
+  private matDialogRef: MatDialogRef<WindowsigninComponent>,
+  private userOwnSignInService: UserOwnSignInService,
+  private router: Router,
+  private authService: AuthService,
+  private googleService: GoogleSignInService,
+  private localStorageService: LocalStorageService,
+  public dialog: MatDialog,
+  @Inject(MAT_DIALOG_DATA) public data: any
+
+  ) { }
+
+    readonly formPic = 'assets/img/formpic.jpg';
+    readonly picGoogle = 'assets/img/icon/google.svg';
+    readonly picArrow = 'assets/img/icon/arrows/google-arrow.svg';
+
+  ngOnInit() {
+  	this.userOwnSignIn = new UserOwnSignIn();
+    this.loadingAnim = false;
+    this.emailErrorMessageBackEnd = null;
+    this.passwordErrorMessageBackEnd = null;
+    this.backEndError = null;
+    this.localStorageService.userIdBehaviourSubject.subscribe(userId => {
+      if (userId) {
+        this.matDialogRef.close();
+      }
+    });
+  }
+
+   signIn(userOwnSignIn: UserOwnSignIn) {
```

Comment:
General: Please add access modifiers to this file and also specify return type for each method

Reply:
what does it mean "access modifiers to this file"? can you explain?

Reply:
You should use "public", "private" depending on situation

Reply:
ready